pkg/curatedpackages/packagecontrollerclient.go

```diff
@@ -0,0 +1,48 @@
+package curatedpackages
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/aws/eks-anywhere-packages/pkg/bundle"
+	"github.com/aws/eks-anywhere/pkg/constants"
+)
+
+type PackageControllerClient struct {
+	kubeConfig     string
+	ociUri         string
+	chartName      string
+	chartVersion   string
+	chartInstaller ChartInstaller
+	kubectl        KubectlRunner
+}
+
+type ChartInstaller interface {
+	InstallChartFromName(ctx context.Context, ociURI, kubeConfig, name, version string) error
+}
+
+func NewPackageControllerClient(chartInstaller ChartInstaller, kubectl KubectlRunner, kubeConfig, ociUri, chartName, chartVersion string) *PackageControllerClient {
+	return &PackageControllerClient{
+		kubeConfig:     kubeConfig,
+		ociUri:         ociUri,
+		chartName:      chartName,
+		chartVersion:   chartVersion,
+		chartInstaller: chartInstaller,
+		kubectl:        kubectl,
+	}
+}
+
+func (pc *PackageControllerClient) InstallController(ctx context.Context) error {
+	uri := fmt.Sprintf("%s%s", "oci://", pc.ociUri)
+	return pc.chartInstaller.InstallChartFromName(ctx, uri, pc.kubeConfig, pc.chartName, pc.chartVersion)
+}
+
+func (pc *PackageControllerClient) ValidateControllerExists(ctx context.Context) error {
+	params := []string{"get", "packageBundleController", "--kubeconfig", pc.kubeConfig, "--namespace", constants.EksaPackagesName, bundle.PackageBundleControllerName}
+	stdOut, _ := pc.kubectl.ExecuteCommand(ctx, params...)
+	if len(stdOut.Bytes()) != 0 {
```

Comment:
What if `kubectl` returns a warning message or some other output that may not be the controller itself? Is this reliable?